### PR TITLE
PUBDEV-7442: Add new "Kubernetes Integration" topic to the welcome.rst

### PR DESCRIPTION
**Pull request by hannah-tillman**

### CODE REVIEW

File: h2o-docs/src/product/welcome.rst [hunk 1]
```diff
@@ -893,3 +893,118 @@ After obtaining the IP address, point your browser to the specified ip address a
...
+                protocol: TCP
+            env:
+            - name: H2O_KUBERNETES_SERVICE_DNS
+              value: h2o-service.h2o-statefulset.svc.cluster.local
+            - name: H2O_NODE_LOOKUP_TIMEOUT
+              value: '180'
+            - name: H2O_NODE_EXPECTED_COUNT
+              value: '3'
+
+There are several other environment variables:
```
Comment (angela0xdata) [inline h2o-docs/src/product/welcome.rst:999]: "Below are additional environment variables:"

File: h2o-docs/src/product/welcome.rst [hunk 2]
```diff
@@ -893,3 +893,118 @@ After obtaining the IP address, point your browser to the specified ip address a
...
+
+- ``H2O_KUBERNETES_SERVICE_DNS`` - **[MANDATORY]** Crucial for the clustering to work: the format usually follows the ``<service-name>.<project-name>.svc.cluster.local`` pattern. This setting enables H2O node discovery via DNS. It must be modified to match the name of the headless service created.
```
Comment (angela0xdata) [inline h2o-docs/src/product/welcome.rst:1001]: - ``H2O_KUBERNETES_SERVICE_DNS`` (Required): Specify the H2O Kubernetes Service DNS to enable H2O node discovery via DNS. This is crucial for the clustering to work. The format usually follows the ``<service-name>.<project-name>.svc.cluster.local`` pattern.  This must be modified to match the name of the headless service created.

File: h2o-docs/src/product/welcome.rst [hunk 3]
```diff
@@ -893,3 +893,118 @@ After obtaining the IP address, point your browser to the specified ip address a
...
+- ``H2O_NODE_LOOKUP_TIMEOUT`` - **[OPTIONAL]** Node lookup constraint: time before the node lookup is ended (defaults to 3 minutes).
```
Comment (angela0xdata) [inline h2o-docs/src/product/welcome.rst:1002]: - ``H2O_NODE_LOOKUP_TIMEOUT`` (Optional): The node lookup constraint. Specify the time before the node lookup times out. (Defaults to 3 minutes.)

File: h2o-docs/src/product/welcome.rst [hunk 4]
```diff
@@ -893,3 +893,118 @@ After obtaining the IP address, point your browser to the specified ip address a
...
+- ``H2O_NODE_EXPECTED_COUNT`` - **[OPTIONAL]** Node lookup constraint: expected number of H2O pods to be discovered.
```
Comment (angela0xdata) [inline h2o-docs/src/product/welcome.rst:1003]: - ``H2O_NODE_EXPECTED_COUNT`` (Optional): The node lookup constraint. This is the expected number of H2O pods to be discovered. This value defaults to 3.

**Comment by angela0xdata** [comment]
@hannah-tillman I'm adding @Pscheidl as a reviewer.

**Comment by Pscheidl** [comment]
Thank you so much @hannah-tillman .

File: h2o-docs/src/product/welcome.rst [hunk 5]
```diff
@@ -893,3 +893,123 @@ After obtaining the IP address, point your browser to the specified ip address a
         # Initialize H2O 
         import h2o
         docker_h2o = h2o.init(ip = "192.168.59.103", port = 54321) 
+
+
+Kubernetes Integration
+----------------------
+
```
Comment (Pscheidl) [inline h2o-docs/src/product/welcome.rst:900]:
```suggestion
H2O nodes must be treated as stateful by Kubernetes environment.  In Kubernetes, a set of pods sharing a common state is named as StatefulSet. H2O is a stateful application. H2O Nodes are spawned together and deallocated together, as a single unit. Kubernetes tooling for stateless applications in not applicable to H2O.
```

Comment (Pscheidl) [reply h2o-docs/src/product/welcome.rst:900]: One small suggestion, I'm not sure it is necessary, as this concept is explained a little bit later. Feel free to discard.

**Pull request merged**